### ISSUE TITLE
Restoring versions reports an error but actually works

### ISSUE THREAD

**Issue by Martchus** [syncthing/syncthing#6253]
### Summary
Restoring deleted file versions via the web UI works but in some cases it wrongly shows an error that it didn't work.

### Steps to reproduce
The wrong error is not shown in all cases and I'm not sure what makes the cases when it is shown special. But if it is happening it is quite simple to reproduce:

1. Select one or more files to restore via the web UI and confirm.
2. An error is shown that renaming didn't work for each file although all of them could be restored¹, e.g.:
    * ![Screenshot_20200105_040758](https://user-images.githubusercontent.com/10248953/71774591-4cd49980-2f72-11ea-964f-caa15da8897e.png)
    * ![Screenshot_20200105_182640](https://user-images.githubusercontent.com/10248953/71783576-fb1b2600-2fe8-11ea-9cb4-273a5d0c1b2b.png)

¹With "could be restored" I mean that the files were under `.stversions` before trying to restore them (Syncthing's versioning put them there as expected). After restoring the files they are again under their old location (as expected).

### Expected behavior
The dialog is just closed unless restoring actually didn't work.

### Further information
The full error message (C&P from the web browser to see the full paths) looks like this:

```
Some items could not be restored:
arch/ownstuff-experimental/os/x86_64/mingw-w64-qt5-3d-5.14.0-1-x86_64.pkg.tar.xz	rename /run/media/devel/repo/.stversions/arch/ownstuff-experimental/os/x86_64/.syncthing.mingw-w64-qt5-3d-5.14.0-1-x86_64.pkg.tar.xz.tmp /run/media/devel/repo/.stversions/arch/ownstuff-experimental/os/x86_64/mingw-w64-qt5-3d-5.14.0-1-x86_64.pkg.tar.xz: no such file or directory
```

So apparently there's some intermediate file called `.syncthing.*.tmp` involved. Not sure whether that's supposed to happen or not. In theory it should be possible to rename the file `/run/media/devel/repo/.stversions/arch/ownstuff-experimental/os/x86_64/mingw-w64-qt5-3d-5.14.0-1-x86_64.pkg.tar.xz` directly, right?

The Syncthing log doesn't contain anything related.

### Further UI issues
* One screenshot looks also darkened as if a modal dialog was still open. However, I could reproduce this problem only once and it is likely just some weird bootstrap problem.
* I'd rather not elide the text of such error messages and instead just wrap it or show a scroll bar. Otherwise one has to C&P the message to get the full information which is quite inconvenient.

### Version Information
Syncthing Version: v1.3.2
Environment: Arch Linux using official Go package, x86_64, ext4 file system

**Comment by AudriusButkevicius** [comment]
Run with STTRACE=versioner env var, and see if there are clues in the log.

**Comment by AudriusButkevicius** [comment]
Folded into #7965
